### PR TITLE
Lesson 6: Mailroom part four with unit testing 

### PR DESCRIPTION
**Pull request by cbrown285**

### CODE REVIEW

File: students/cbrown/Lesson 6/mailroom_part_four.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+#Version B- Focusing on send_thanks refactoring
+#!/usr/bin/env python3
+#Mailroom Script Part 4
+#Refactoring
+import sys
+
+donors = {"Bill Gates":[539000,235642],
```
Comment (charlesrobison) [inline students/cbrown/Lesson 6/mailroom_part_four.py:7]: Missing whitespaces after colon and comma - here and below

File: students/cbrown/Lesson 6/mailroom_part_four.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+          "Mark Cuban":[459035,369.50,570.89]}
+
+prompt = "\n".join(("Please Select from Items Below:",
+                    "1 - Send A Thank You",
+                    "2 - Create A Report",
+                    "3 - Send Letters to All Donors",
+                    "4 - Quit",
+                    " "))
+
+def donor_verify(donor_name,amount,donor_list):
```
Comment (charlesrobison) [inline students/cbrown/Lesson 6/mailroom_part_four.py:20]: Missing space after comma

File: students/cbrown/Lesson 6/mailroom_part_four.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+    #not found in list. Add new donor entry
+    else:
+        new_entry = []
+        new_entry.append(amount)
+        donors[donor_name] = new_entry
+
+    return donors
+
+def thank_you_note(donor_name,amount):
```
Comment (charlesrobison) [inline students/cbrown/Lesson 6/mailroom_part_four.py:34]: Here too - as well as in other spots below.

File: students/cbrown/Lesson 6/mailroom_part_four.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    Prints the output of the donor report using data generated from report_rows function & report_lines list
+    '''
+    #prints the header of the report
+    print('{:<25}{}{:^15}{}{:^15}{}{:^15}'.format('Donor Name','|','Total Given','|','Num Gifts','|','Average Gift'))
+    str_len = len('{:<25}{}{:^15}{}{:^15}{}{:^15}'.format('Donor Name','|','Total Given','|','Num Gifts','|','Average Gift'))
+    print('-' * str_len)
+
+    #prints name, total, num of gifts, and avg gift
+    for row in report_lines:
+        print(row)
```
Comment (charlesrobison) [inline students/cbrown/Lesson 6/mailroom_part_four.py:107]: Report is not picking up new donors and donations

**Review by charlesrobison — APPROVED** [review]
Tests look good, only missing new donors in report.  Also, recommend adding a linter to your text editor if you don't have one already.  It will catch the missing space issues.

Ready to merge

File: students/cbrown/Lesson 6/mailroom_part_four.py (hunk 4 above)
Comment (cbrown285) [reply students/cbrown/Lesson 6/mailroom_part_four.py:104]: Fixed this and resubmitted!

**Review by charlesrobison — APPROVED** [review]
Updates look good - nice job.  Updating grade.

Ready to merge.

**Pull request merged**